src/main/java/com/gregtechceu/gtceu/client/renderer/monitor/MonitorTextRenderer.java

```diff
@@ -34,6 +34,23 @@ public void render(CentralMonitorMachine machine, MonitorGroup group, float part
             int row = 0;
             int columns = group.getRow(0, machine::toRelative).size();
             poseStack.translate(rel.getX(), rel.getY(), rel.getZ());
+            int layer = 0;
+            for (GraphicsComponent graphics : text.getGraphics()) {
+                if (graphics.x() < 0 || graphics.y() < 0) continue;
+                double maxX = graphics.x2();
+                double maxY = graphics.y2();
+                if (maxX == Math.floor(maxX)) maxX--;
+                if (maxY == Math.floor(maxY)) maxY--;
+                BlockPos relativePos = rel.mutable().move((int) Math.floor(maxX), (int) -Math.floor(maxY), 0);
+                if (!group.getMonitorPositions().stream().map(machine::toRelative).toList().contains(relativePos))
+                    continue;
+                poseStack.pushPose();
+                poseStack.translate(graphics.x(), graphics.y(), layer * .001);
```

Comment:
Are you adding the layer offset twice intentionally?
```suggestion
                poseStack.translate(graphics.x(), graphics.y(), layer * 0.001f);
```

Reply:
i'm adding it only once (note the popPose call after the first translate with layer offset)